extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,263 @@
+import org.apache.rya.streams.api.queries.QueryChange;
+import org.apache.rya.streams.api.queries.QueryChange.ChangeType;
+import org.apache.rya.streams.api.queries.QueryChangeLog;
+import org.apache.rya.streams.api.queries.QueryChangeLogListener;
+import org.apache.rya.streams.api.queries.QueryRepository;
+import org.apache.rya.streams.querymanager.QueryChangeLogSource.SourceListener;
+import org.apache.rya.streams.querymanager.QueryExecutor.QueryExecutorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.AbstractScheduledService.Scheduler;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A service for managing {@link StreamsQuery} running on a Rya Streams system.
+ * <p>
+ * Only one QueryManager needs to be running to manage any number of rya
+ * instances/rya streams instances.
+ */
+@DefaultAnnotation(NonNull.class)
+public class QueryManager extends AbstractIdleService {
+    private static final Logger LOG = LoggerFactory.getLogger(QueryManager.class);
+
+    private final QueryExecutor queryExecutor;
+    private final Scheduler scheduler;
+
+    /**
+     * Map of Rya Instance name to {@link QueryRepository}.
+     */
+    private final Map<String, QueryRepository> queryRepos;
+
+    private final QueryChangeLogSource source;
+
+    private final ReentrantLock lock;
+
+    /**
+     * Creates a new {@link QueryManager}.
+     *
+     * @param queryExecutor - The {@link QueryExecutor} that starts/creates
+     *        queries when a CREATED query is found in the
+     *        {@link QueryChangeLog}. (not null)
+     * @param source - The {@link QueryChangeLogSource} of QueryChangeLogs. (not null)
+     * @param scheduler - The {@link Scheduler} to use throughout
+     */
+    public QueryManager(final QueryExecutor queryExecutor, final QueryChangeLogSource source, final Scheduler scheduler) {
+        this.source = requireNonNull(source);
+        this.queryExecutor = requireNonNull(queryExecutor);
+        this.scheduler = requireNonNull(scheduler);
+
+        queryRepos = new HashMap<>();
+
+        lock = new ReentrantLock();
+    }
+
+    /**
+     * Starts running a query.
+     *
+     * @param ryaInstanceName - The Rya instance the query belongs to.
+     * @param query - The query to run.(not null)
+     */
+    private void runQuery(final String ryaInstanceName, final StreamsQuery query) {
+        requireNonNull(query);
+        LOG.info("Starting Query: " + query.getSparql());
+        if (!queryExecutor.isRunning()) {
+            throw new IllegalStateException("Query Executor must be started before queries can be managed.");
+        }
+
+        try {
+            queryExecutor.startQuery(ryaInstanceName, query);
+        } catch (final QueryExecutorException e) {
+            LOG.error("Failed to start query.", e);
```

Comment:
So we just give up and never try to start it again? I'm not sure what we should be doing here.

Reply:
I mean we could have a thread that occasionally tries to start it until it succeeds, but that also doesn't seem right.  The biggest problem is that if this does fail and logs, if no one is watching the logs there's no way to know it failed.  A claxon?

Reply:
I think we're kind of covered by having the QueryExecutor be responsible for managing what is started and stopped. So really, it throwing the exception indicates it wasn't able to even process the start request. But we haven't really formalized that yet beyond being able to ask the executor what is started. Maybe we should think a little more on what the executor's responsibilities are.